recipes/frugen/all/patches/0003-let-conan-choose-optimization-flags.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 763c0b3..0ce32df 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -39,8 +39,8 @@ option(ENABLE_JSON "enable JSON support" ON)
+ option(JSON_STATIC "link json-c library statically" OFF)
+ option(DEBUG_OUTPUT "show extra debug output" OFF)
+ 
+-set(CMAKE_C_FLAGS_RELEASE "-Os -Wall -Werror -Wfatal-errors")
+-set(CMAKE_C_FLAGS_DEBUG "-g3 -O0 -Wall -Werror -Wfatal-errors")
++# set(CMAKE_C_FLAGS_RELEASE "-Os -Wall -Werror -Wfatal-errors")
```

Comment:
@alatarum I removed this as Conan usually injects these flags based on the build_type provided in the profile, so that users can better select their desired optimization. Is -Os in this case critical?

Reply:
Thanks for testing and improving!
I almost sure that -Os is just a personal preference of the author and not actually needed. I will look is we should remove this lines in upstream also.

BTW is it possible to ask conan to provide PkgConfig files for json-c when this recipe pulls it as a dependency? Then first patch not needed.

Reply:
using `PkgConfigDeps` class would allow that to work, but trying locally, then I get some include errors for other files, which confuses me a bit:

You can try locally with:

```diff
diff --git a/recipes/frugen/all/conandata.yml b/recipes/frugen/all/conandata.yml
index cc18fdc79..1474da704 100644
--- a/recipes/frugen/all/conandata.yml
+++ b/recipes/frugen/all/conandata.yml
@@ -4,9 +4,6 @@ sources:
     sha256: "cb5c59cf3593b162b9faedf1e5ae3b1f7e32fb732a61ada6baf7de69632bff77"
 patches:
   "2.0":
-    - patch_file: "patches/0001-cmake-use-find_package-to-locate-json-c-library.patch"
-      patch_description: "Upstream prefer to use pkg_check_modules instead of find_package to keep capatibility with old versions"
-      patch_type: "conan"
     - patch_file: "patches/0002-libfru-fix-build-on-MacOS.patch"
       patch_description: "Fixes build on MacOS"
       patch_type: "portability"
diff --git a/recipes/frugen/all/conanfile.py b/recipes/frugen/all/conanfile.py
index 17b6eeec9..b2becc771 100644
--- a/recipes/frugen/all/conanfile.py
+++ b/recipes/frugen/all/conanfile.py
@@ -1,6 +1,7 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.gnu import PkgConfigDeps
 from conan.tools.files import get, rmdir, apply_conandata_patches, export_conandata_patches
 from conan.tools.microsoft import is_msvc
 import os
@@ -65,6 +66,8 @@ class FrugenConan(ConanFile):
         tc.generate()
         deps = CMakeDeps(self)
         deps.generate()
+        pkgdeps = PkgConfigDeps(self)
+        pkgdeps.generate()
 
     def build(self):
         cmake = CMake(self)
```

Reply:
Hmm, I'm getting linking error:
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -ljson-c: No such file or directory

But let's keep current implementation

Reply:
We can go for a follow-up PR once we find how to properly use it there, but for now this looks good!